pkg/gce-cloud-provider/compute/gce.go

```diff
@@ -192,15 +193,24 @@ func newOauthClient(tokenSource oauth2.TokenSource) (*http.Client, error) {
 	return oauth2.NewClient(context.Background(), tokenSource), nil
 }
 
-func getProjectAndZoneFromMetadata() (string, string, error) {
+func getProjectAndZone(config *ConfigFile) (string, string, error) {
+	var err error
+
 	zone, err := metadata.Zone()
 	if err != nil {
 		return "", "", err
 	}
-	projectID, err := metadata.ProjectID()
-	if err != nil {
-		return "", "", err
+
+	var projectID string
+	if config == nil || config.Global.ProjectId == "" {
```

Comment:
should we spit out a warning when this happens? Is this expected behavior? 

If `config != nil` I would sort of expect the project ID being empty to be an error or something. If thats not a case could you write a comment explaining how we could get into a situation like that?

It just feels like a case where we could potentially want to use the `config`'s project ID and we'd have a silent failure here if that ever had any problems.

Reply:
ACK on the warning. I'll also add a comment to explain things